Sources/SwiftProtobuf/BinaryDecoder.swift

```diff
@@ -892,13 +892,13 @@ internal struct BinaryDecoder: Decoder {
         if !subdecoder.complete {
             throw BinaryDecodingError.trailingGarbage
         }
-
-        if let k = k, let v = v {
-            value[k] = v
-            consumed = true
-        } else {
-            throw BinaryDecodingError.malformedProtobuf
-        }
+        // A map<> definition can't provide a default value for the keys/values,
+        // so it is safe to use the proto3 default to get the right
+        // integer/string/bytes. The one catch is a proto2 enum (which can be the
+        // value) can have a non zero value, but that case is the next
+        // custom decodeMapField<>() method and handles it.
+        value[k ?? KeyType.proto3DefaultValue] = v ?? ValueType.proto3DefaultValue
```

Comment:
The other option is to make k and v non optional, but that ripples through all the MapKey/MapValue types as there is no decodeSingle that takes a non option, can easily be added, but I wasn't sure if it was worth it.